back_end/README.md

```diff
@@ -1,21 +1,83 @@
-- [ ] Add another endpoint to your service that allows for the bulk (hundreds) submission of URLs.
-- [ ] Your submission should contain a README file that spells out the exact steps necessary to build and run the project. It should also provide instructions for how to query the webservice.
+- [ ] You can select the language you think is the best for this challenge. If you decide to use a functional programming like elixir just let us know first.
+- [ ] You are free to add any library for improving the development experience, like, linters, formatters, etc. Try to emulate the environment that you would like to have when you are working.
+- [ ] You should design your APi for evolvability. Given that in real life no API stays static for long.  
+- [ ] You can use any API paradigm. Rest, RPC or GraphQL depending on what you think is best for the challenge. 
+- [ ] The app should have the tests that you decide are necessary for maintainability
+- [ ] You should handle all success and errors cases the endpoints could have. When there is an error the backend should notify the client about it in such a way that allows the client side to decide what to do. (Ex. If the client tries to create an article without a title we expect the API to notify the user that the title is a required attribute in order to create an article) 
+- [ ] There should be a way for client users to discover how to use the API. It can be a document, a postman, a swagger document or a plain html website. Is up to you but it should allow the user to discover all your endpoints, and their possible responses both success and errors. 
 
-Design of the API endpoint including paths, structure, and format is up to you. Decisions on how to structure the code and what frameworks to use are up to you as well.
+## Bonus Points 
 
-## Bonus
+Take 1 or 2, we don't expect for you to do everyone of them. 
 
-- [ ] Build a frontend for the service in any frontend framework that allows users to submit a URL to shorten it in a form, and see the result.
-- [ ] Extend your frontend to allow a text file full of URLs to be submitted to the bulk API. Results can be rendered directly on the page or downloaded as a text file.
-- [ ] Add a view to list all the URLs and their short versions.
+- [ ] Update the list of user stories, and the API documentation if we add a new actor called Admin that is the only one with the permission to publish articles but only when an article is "ready to review". Think about how that new actor will change your current implementation and what changes you will need to do.
+- [ ] Deploy the API in a Platform as a Service (Heroku, Blue Ocean, AWS, GCloud, etc)
+- [ ] Dockerize the development environment so the database, programming language version, etc lives in a container. The idea is that when a new developer starts on the project she will not have to configure the development environment by hand. 
```

Comment:
`when a new developer starts on the project she will not` => `when new developers starts on the project they will not`